app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_details/ReminderType.kt

```diff
@@ -1,9 +1,44 @@
 package com.humberto.tasky.agenda.presentation.agenda_details
 
+import com.humberto.tasky.R
+import com.humberto.tasky.agenda.domain.AgendaItem
+import com.humberto.tasky.core.presentation.ui.UiText
+import java.time.Duration
+import java.time.ZonedDateTime
+
 enum class ReminderType {
     TenMinutes,
     ThirtyMinutes,
     OneHour,
     SixHours,
     OneDay
+}
+
+fun AgendaItem.getReminderType(): ReminderType? {
+    val minutesDifference = Duration.between(from, remindAt).toMinutes()
+    return ReminderType.entries.find { it.toReminderMinutes() == minutesDifference }
+}
+
+private fun ReminderType.toReminderMinutes(): Long {
+    return when (this) {
+        ReminderType.TenMinutes -> 10L
+        ReminderType.ThirtyMinutes -> 30L
+        ReminderType.OneHour -> 60L
+        ReminderType.SixHours -> 360L
+        ReminderType.OneDay -> 1440L
+    }
+}
```

Comment:
Be careful with primitive types like `Long`. Generally, when using a `Long` in relation to time, it represents milliseconds. Here, you are using it to represent minutes. To better communicate this and avoid confusion, use a type alias.

```
typealias Minutes = Long
```

Your function can then return this:

```
private fun ReminderType.toReminderMinutes(): Minutes {
    return when (this) {
        ReminderType.TenMinutes -> 10L
        ReminderType.ThirtyMinutes -> 30L
        ReminderType.OneHour -> 60L
        ReminderType.SixHours -> 360L
        ReminderType.OneDay -> 1440L
    }
}
```